### PR TITLE
Composer require checker

### PR DESCRIPTION
**Pull request by greg0ire**
This tool browses the project and report things that are used but not referenced directly. There are more of them:

```
+-----------------------------------------------------------------+--------------------+
| unknown symbol                                                  | guessed dependency |
+-----------------------------------------------------------------+--------------------+
| iconv                                                           | ext-iconv          |
| Doctrine\Common\Persistence\ManagerRegistry                     |                    |
| Doctrine\ODM\PHPCR\DocumentManager                              |                    |
| Doctrine\ORM\EntityManagerInterface                             |                    |
| JMS\Serializer\Handler\SubscribingHandlerInterface              |                    |
| PHPUnit\Framework\TestCase                                      |                    |
| Doctrine\ORM\AbstractQuery                                      |                    |
| Doctrine\ORM\Version                                            |                    |
| Doctrine\ORM\QueryBuilder                                       |                    |
| Doctrine\ORM\EntityRepository                                   |                    |
| Doctrine\Common\Persistence\Mapping\ClassMetadata               |                    |
| Doctrine\ORM\EntityManager                                      |                    |
| Symfony\Bridge\Twig\Form\TwigRenderer                           |                    |
| Symfony\Bridge\Doctrine\Form\Type\EntityType                    |                    |
| Nelmio\ApiDocBundle\Form\Extension\DescriptionFormTypeExtension |                    |
| Symfony\Component\OptionsResolver\OptionsResolverInterface      |                    |
| Locale                                                          | ext-intl           |
| IntlDateFormatter                                               | ext-intl           |
| Metadata\MetadataFactoryInterface                               |                    |
| Symfony\Component\Console\Input\InputOption                     |                    |
| Symfony\Component\Console\Input\InputInterface                  |                    |
| Symfony\Component\Console\Output\OutputInterface                |                    |
| Gaufrette\Adapter\Local                                         |                    |
| Gaufrette\Filesystem                                            |                    |
| json_encode                                                     | ext-json           |
| Doctrine\ORM\Mapping\ClassMetadata                              |                    |
| Exporter\Source\SourceIteratorInterface                         |                    |
| Exporter\Writer\XlsWriter                                       |                    |
| Exporter\Writer\XmlWriter                                       |                    |
| Exporter\Writer\JsonWriter                                      |                    |
| Exporter\Writer\CsvWriter                                       |                    |
| Exporter\Handler                                                |                    |
| Symfony\Component\Validator\ExecutionContextInterface           |                    |
| JMS\Serializer\GraphNavigator                                   |                    |
| JMS\Serializer\VisitorInterface                                 |                    |
| JMS\Serializer\Context                                          |                    |
+-----------------------------------------------------------------+--------------------+
```

But that corresponds to optional deps, or interfaces/classes that exist only in old versions

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -20,9 +20,13 @@
         "cocur/slugify": "^1.4 || ^2.0 || ^3.0",
         "sonata-project/datagrid-bundle": "^2.0",
         "symfony/config": "^2.8 || ^3.2 || ^4.0",
+        "symfony/event-dispatcher": "^2.8 || ^3.2 || ^4.0",
+        "symfony/dependency-injection": "^2.8 || ^3.2 || ^4.0",
```
Comment (jordisala1991) [inline composer.json]: sort this please :)

Comment (greg0ire) [reply composer.json]: woops

**Comment by core23** [comment]
Which tool are you using for this?

**Comment by greg0ire** [comment]
@core23 https://github.com/maglnet/ComposerRequireChecker/ , it's awesome

**Comment by core23** [comment]
Thanks :)

**Pull request merged**